src/main/conversions.c

```diff
@@ -877,10 +878,15 @@ as_status pyobject_to_val(AerospikeClient *self, as_error *err,
         PyObject *py_parameter = PyString_FromString("geo_data");
         PyObject *py_data = PyObject_GenericGetAttr(py_obj, py_parameter);
         Py_DECREF(py_parameter);
-        char *geo_value =
-            PyString_AsString(AerospikeGeospatial_DoDumps(py_data, err));
+
+        PyObject *geospatial_dump = AerospikeGeospatial_DoDumps(py_data, err);
+        char *geo_value = PyString_AsString(geospatial_dump);
+        char *geo_value_cpy = strdup(geo_value);
+
         Py_DECREF(py_data);
-        *val = (as_val *)as_geojson_new(geo_value, false);
+        Py_DECREF(geospatial_dump);
+
+        *val = (as_val *)as_geojson_new(geo_value_cpy, true);
```

Comment:
Just curious if this was causing any observable issues? Was geospatial_dump never being cleaned up?

Reply:
This change doesn't seem to fix anything; I thought it did when I compared the valgrind logs before and after this change. There is still a leak from calling `AerospikeGeospatial_DoDumps()` in line 882 that hasn't been fixed.

Stage branch: https://github.com/aerospike/aerospike-client-python/actions/runs/4515075980/jobs/7951932957
This branch: https://github.com/aerospike/aerospike-client-python/actions/runs/4622113868/jobs/8174383185

I'm not seeing any additional memory leaks or errors that were introduced by this change.

Even so, I think the change where we pass in a copy of the string to `as_geojson_new()` is correct, since `PyString_AsString()` (which is actually [PyUnicode_AsUTF8()](https://github.com/aerospike/aerospike-client-python/blob/CLIENT-2159-cicd-valgrind/src/include/macros.h#L34)) is returning a buffer that gets deallocated when the PyUnicode object `geospatial_dump` is garbage collected, and we don't want the as_geojson object to point to invalid memory once that happens.

Reply:
You should look into fixing the leak you are seeing in AerospikeGeospatial_DoDumps() when you can